bin/auth_ejabberd.php

```diff
@@ -52,8 +53,7 @@
 
 require dirname(__DIR__) . '/vendor/autoload.php';
 
-$dice = new \Dice\Dice();
-$dice = $dice->addRules(include __DIR__ . '/../static/dependencies.config.php');
+$dice = (new Dice())->addRules(include __DIR__ . '/../static/dependencies.config.php');
```

Comment:
Can you please replace all calls to Dice->addRules by adding the rules directly in the constructor?

What would be the benefit of explicitly calling addRules?

Reply:
The `Dice` constructor has no parameters. The immutable pattern there is to create a "fresh" new instance without loaded configs.

See an explanation for DICE here:
https://github.com/Level-2/Dice/issues/166#issuecomment-478351720

Reply:
What about this then: https://github.com/friendica/friendica/pull/7465/commits/2e65b96cb4b0193edcf6c0e6a4647f7ebbe01bb8#diff-be1573d8ce2c3198f956d1f0c8e654d0R90

Reply:
This is an error ;) and I now see that the IDE 'cries' about it. But it seems that PHP is error friendly, so the constructor just ignores it ;)

Reply:
Then please modify this instance accordingly.

Reply:
Done :D